### PR TITLE
Disable minification in production to get better stacktraces

### PR DESCRIPTION
**Pull request by lukaw3d**
Fixes https://github.com/oasisprotocol/oasis-wallet-web/issues/495

### CODE REVIEW

File: internals/scripts/build-without-minify.js [hunk 1]
```diff
@@ -23,5 +23,5 @@ proxyquire('react-scripts/scripts/build.js', {
     webpackConfig.optimization.minimize = false
 
     return webpackConfig
-  },
+  }
```
Comment (lukaw3d) [inline internals/scripts/build-without-minify.js:26]: :facepalm: Mega-Linter vs Prettier

**Review by pro-wh — APPROVED** [review]
proxyquire sounds like it's a pretty big hammer

if we're fiddling with the webpack config, could we instead have it make source maps?

**Comment by pro-wh** [comment]
aw ):

File: internals/scripts/build-without-minify.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * If this ever starts breaking: just switch back to `build:minified`.
+ */
+
+const proxyquire = require('proxyquire')
+
+proxyquire('react-scripts/scripts/build.js', {
+  '../config/webpack.config': webpackEnv => {
+    const configFactory = require('react-scripts/config/webpack.config')
+    const webpackConfig = configFactory(webpackEnv)
```
Comment (lukaw3d) [inline internals/scripts/build-without-minify.js:20]: @pro-wh here's a way without proxyquire

```js
function overrideWebpackConfigModule(originalModule) {
  /** @type {import('react-scripts/config/webpack.config')} */
  const configFactory = originalModule
  return function (webpackEnv) {
    const webpackConfig = configFactory(webpackEnv)
    webpackConfig.optimization.minimize = false
    return webpackConfig
  }
}

// Override global `require`
const Module = require('module')
const originalGlobalRequire = Module.prototype.require
// @ts-ignore
Module.prototype.require = function (filename) {
  const importedModule = originalGlobalRequire.call(this, filename)
  if (filename === '../config/webpack.config') return overrideWebpackConfigModule(importedModule)
  return importedModule
}

require('react-scripts/scripts/build.js')

```

**Comment by matevz** [comment]
I'm not sure if it's a good idea to remove minification by default. Should we rather have the `yarn build` and `yarn build:debug` actions instead. And the first one would then be used on `wallet.oasisprotocol.org` while the latter one on `wallet.stg.oasisprotocol.org`?

**Comment by lukaw3d** [comment]
@matevz How come?
It won't impact performance that much - GetGenesisDocument is much slower than loading 300KB larger javascript.
Obfuscation-wise it is an opensource project.
And bug reports will be much better - users aren't reporting stack-traces from staging

**Comment by mitar** [comment]
Minimization is overrated. We have primarily to assure we use gzip compression on HTTP. And that resources can be cached well.

**Comment by lukaw3d** [comment]
Stacktrace for comparison:

minified:
```js
Http response at 400 or 500 level
The above error occurred in task vr
    created by vr
    created by takeEvery(account/fetchAccount, vr)
    created by xr
Tasks cancelled due to error:
takeEvery(account/fetchAccount, vr)
Or

Error: Http response at 400 or 500 level
    at new D (https://wallet.oasisprotocol.org/static/js/2.48cea8b1.chunk.js:2:1047878)
    at Ft (https://wallet.oasisprotocol.org/static/js/2.48cea8b1.chunk.js:2:1070696)
    at _t.<anonymous> (https://wallet.oasisprotocol.org/static/js/2.48cea8b1.chunk.js:2:1069926)
    at Ue (https://wallet.oasisprotocol.org/static/js/2.48cea8b1.chunk.js:2:1056709)
    at Be (https://wallet.oasisprotocol.org/static/js/2.48cea8b1.chunk.js:2:1056515)
    at xt (https://wallet.oasisprotocol.org/static/js/2.48cea8b1.chunk.js:2:1064004)
    at Tt (https://wallet.oasisprotocol.org/static/js/2.48cea8b1.chunk.js:2:1064785)
    at _t.r.W (https://wallet.oasisprotocol.org/static/js/2.48cea8b1.chunk.js:2:1065726)
    at _t.r.R (https://wallet.oasisprotocol.org/static/js/2.48cea8b1.chunk.js:2:1065705)
```

unminified:
```js
Http response at 400 or 500 level
The above error occurred in task loadAccount
    created by loadAccount
    created by takeEvery(account/fetchAccount, loadAccount)
    created by accountSaga
Tasks cancelled due to error:
anonymous
anonymous
takeEvery(account/fetchAccount, loadAccount)
refreshAccountOnTransaction

Error: Http response at 400 or 500 level
    at new E (http://localhost:3000/static/js/vendors~main.chunk.js:161676:13)
    at Y (http://localhost:3000/static/js/vendors~main.chunk.js:163163:33)
    at oc.<anonymous> (http://localhost:3000/static/js/vendors~main.chunk.js:163137:25)
    at Ib (http://localhost:3000/static/js/vendors~main.chunk.js:162302:22)
    at O (http://localhost:3000/static/js/vendors~main.chunk.js:162288:3)
    at wc (http://localhost:3000/static/js/vendors~main.chunk.js:162797:21)
    at yc (http://localhost:3000/static/js/vendors~main.chunk.js:162864:9)
    at oc.push../node_modules/grpc-web/index.js.n.W (http://localhost:3000/static/js/vendors~main.chunk.js:162809:3)
    at oc.push../node_modules/grpc-web/index.js.n.R (http://localhost:3000/static/js/vendors~main.chunk.js:162805:59)
```

Bottom of stacktrace is still minified because https://www.npmjs.com/package/grpc-web ships pre-minified
```js
//    at new E (http://localhost:3000/static/js/vendors~main.chunk.js:161676:13):
function E(a, b, c) {
  c = void 0 === c ? {} : c;
  b = Error.call(this, b);
  this.message = b.message;
  "stack" in b && (this.stack = b.stack);
  this.code = a;
  this.metadata = c;
}

oa(E, Error);
E.prototype.name = "RpcError";
```

**Comment by pro-wh** [comment]
I think we don't need the frames inside grpc-web. They're pretty good with having unique error messages for different situations. It's that we don't have any async stack from outside of grpc-web's response handler.

**Comment by lukaw3d** [comment]
@pro-wh found that Firefox doesn't cut stacktrace

```js
Error: Http response at 400 or 500 level

The above error occurred in task loadAccount
    created by loadAccount
    created by takeEvery(account/fetchAccount, loadAccount)
    created by accountSaga
Tasks cancelled due to error:
anonymous
anonymous
takeEvery(account/fetchAccount, loadAccount)
refreshAccountOnTransaction

E@http://localhost:3000/static/js/vendors~main.chunk.js:161676:13
Y@http://localhost:3000/static/js/vendors~main.chunk.js:163163:33
X/<@http://localhost:3000/static/js/vendors~main.chunk.js:163137:25
Ib@http://localhost:3000/static/js/vendors~main.chunk.js:162302:22
O@http://localhost:3000/static/js/vendors~main.chunk.js:162288:5
wc@http://localhost:3000/static/js/vendors~main.chunk.js:162797:21
yc@http://localhost:3000/static/js/vendors~main.chunk.js:162864:11
./node_modules/grpc-web/index.js/</n.W@http://localhost:3000/static/js/vendors~main.chunk.js:162809:5
./node_modules/grpc-web/index.js/</n.R@http://localhost:3000/static/js/vendors~main.chunk.js:162805:59
EventHandlerNonNull*sc@http://localhost:3000/static/js/vendors~main.chunk.js:162747:28
Nc@http://localhost:3000/static/js/vendors~main.chunk.js:163340:5
./node_modules/grpc-web/index.js/</Z.prototype.S/</<@http://localhost:3000/static/js/vendors~main.chunk.js:163223:17
./node_modules/grpc-web/index.js/</Z.prototype.S/<@http://localhost:3000/static/js/vendors~main.chunk.js:163222:12
./node_modules/grpc-web/index.js/</Z.prototype.S@http://localhost:3000/static/js/vendors~main.chunk.js:163231:14
callUnary@http://localhost:3000/static/js/vendors~main.chunk.js:3839:24
stakingAccount@http://localhost:3000/static/js/vendors~main.chunk.js:4235:17
runCallEffect@http://localhost:3000/static/js/vendors~main.chunk.js:10934:21
runEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11615:19
digestEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11682:19
next@http://localhost:3000/static/js/vendors~main.chunk.js:11572:21
proc@http://localhost:3000/static/js/vendors~main.chunk.js:11519:3
runEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11610:11
digestEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11682:19
runAllEffect/<@http://localhost:3000/static/js/vendors~main.chunk.js:11082:17
runAllEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11081:8
runEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11615:19
digestEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11682:19
next@http://localhost:3000/static/js/vendors~main.chunk.js:11572:21
currCb@http://localhost:3000/static/js/vendors~main.chunk.js:11662:9
runForkEffect/<@http://localhost:3000/static/js/vendors~main.chunk.js:11001:11
immediately@http://localhost:3000/static/js/vendors~main.chunk.js:10455:12
runForkEffect@http://localhost:3000/static/js/vendors~main.chunk.js:10993:14
runEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11615:19
digestEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11682:19
next@http://localhost:3000/static/js/vendors~main.chunk.js:11572:21
currCb@http://localhost:3000/static/js/vendors~main.chunk.js:11662:9
runForkEffect/<@http://localhost:3000/static/js/vendors~main.chunk.js:11001:11
immediately@http://localhost:3000/static/js/vendors~main.chunk.js:10455:12
runForkEffect@http://localhost:3000/static/js/vendors~main.chunk.js:10993:14
runEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11615:19
digestEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11682:19
next@http://localhost:3000/static/js/vendors~main.chunk.js:11572:21
currCb@http://localhost:3000/static/js/vendors~main.chunk.js:11662:9
end@http://localhost:3000/static/js/vendors~main.chunk.js:11423:10
addTask/task.cont@http://localhost:3000/static/js/vendors~main.chunk.js:11244:15
next@http://localhost:3000/static/js/vendors~main.chunk.js:11581:18
currCb@http://localhost:3000/static/js/vendors~main.chunk.js:11662:9
runSelectEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11140:7
runEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11615:19
digestEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11682:19
next@http://localhost:3000/static/js/vendors~main.chunk.js:11572:21
proc@http://localhost:3000/static/js/vendors~main.chunk.js:11519:3
runCallEffect@http://localhost:3000/static/js/vendors~main.chunk.js:10943:11
runEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11615:19
digestEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11682:19
next@http://localhost:3000/static/js/vendors~main.chunk.js:11572:21
currCb@http://localhost:3000/static/js/vendors~main.chunk.js:11662:9
promise callback*resolvePromise@http://localhost:3000/static/js/vendors~main.chunk.js:10806:11
runCallEffect@http://localhost:3000/static/js/vendors~main.chunk.js:10937:21
runEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11615:19
digestEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11682:19
next@http://localhost:3000/static/js/vendors~main.chunk.js:11572:21
currCb@http://localhost:3000/static/js/vendors~main.chunk.js:11662:9
end@http://localhost:3000/static/js/vendors~main.chunk.js:11423:10
addTask/task.cont@http://localhost:3000/static/js/vendors~main.chunk.js:11244:15
next@http://localhost:3000/static/js/vendors~main.chunk.js:11581:18
currCb@http://localhost:3000/static/js/vendors~main.chunk.js:11662:9
runSelectEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11140:7
runEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11615:19
digestEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11682:19
next@http://localhost:3000/static/js/vendors~main.chunk.js:11572:21
proc@http://localhost:3000/static/js/vendors~main.chunk.js:11519:3
runCallEffect@http://localhost:3000/static/js/vendors~main.chunk.js:10943:11
runEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11615:19
digestEffect@http://localhost:3000/static/js/vendors~main.chunk.js:11682:19
next@http://localhost:3000/static/js/vendors~main.chunk.js:11572:21
currCb@http://localhost:3000/static/js/vendors~main.chunk.js:11662:9
runPutEffect/<@http://localhost:3000/static/js/vendors~main.chunk.js:10892:9
exec@http://localhost:3000/static/js/vendors~main.chunk.js:10429:5
flush@http://localhost:3000/static/js/vendors~main.chunk.js:10487:9
asap@http://localhost:3000/static/js/vendors~main.chunk.js:10444:5
stdChannel/chan.put@http://localhost:3000/static/js/vendors~main.chunk.js:10786:9
sagaMiddleware/</<@http://localhost:3000/static/js/vendors~main.chunk.js:11824:17
createSerializableStateInvariantMiddleware/</</<@http://localhost:3000/static/js/vendors~main.chunk.js:13269:26
createThunkMiddleware/</</<@http://localhost:3000/static/js/vendors~main.chunk.js:240005:16
createImmutableStateInvariantMiddleware/</</<@http://localhost:3000/static/js/vendors~main.chunk.js:13154:36
AccountPage/<@http://localhost:3000/static/js/main.chunk.js:23871:13
invokePassiveEffectCreate@http://localhost:3000/static/js/vendors~main.chunk.js:219376:24
callCallback@http://localhost:3000/static/js/vendors~main.chunk.js:199992:18
invokeGuardedCallbackDev@http://localhost:3000/static/js/vendors~main.chunk.js:200041:20
invokeGuardedCallback@http://localhost:3000/static/js/vendors~main.chunk.js:200101:35
flushPassiveEffectsImpl@http://localhost:3000/static/js/vendors~main.chunk.js:219458:34
unstable_runWithPriority@http://localhost:3000/static/js/vendors~main.chunk.js:242802:16
runWithPriority$1@http://localhost:3000/static/js/vendors~main.chunk.js:207398:14
flushPassiveEffects@http://localhost:3000/static/js/vendors~main.chunk.js:219335:18
performSyncWorkOnRoot@http://localhost:3000/static/js/vendors~main.chunk.js:218175:7
flushSyncCallbackQueueImpl/<@http://localhost:3000/static/js/vendors~main.chunk.js:207452:30
unstable_runWithPriority@http://localhost:3000/static/js/vendors~main.chunk.js:242802:16
runWithPriority$1@http://localhost:3000/static/js/vendors~main.chunk.js:207398:14
flushSyncCallbackQueueImpl@http://localhost:3000/static/js/vendors~main.chunk.js:207447:30
flushSyncCallbackQueue@http://localhost:3000/static/js/vendors~main.chunk.js:207435:7
unbatchedUpdates@http://localhost:3000/static/js/vendors~main.chunk.js:218346:11
legacyRenderSubtreeIntoContainer@http://localhost:3000/static/js/vendors~main.chunk.js:221855:25
render@http://localhost:3000/static/js/vendors~main.chunk.js:221938:14
./src/index.tsx/<@http://localhost:3000/static/js/main.chunk.js:52196:40
./src/index.tsx@http://localhost:3000/static/js/main.chunk.js:52318:30
__webpack_require__@http://localhost:3000/static/js/bundle.js:857:31
fn@http://localhost:3000/static/js/bundle.js:151:20
1@http://localhost:3000/static/js/main.chunk.js:62070:18
__webpack_require__@http://localhost:3000/static/js/bundle.js:857:31
checkDeferredModules@http://localhost:3000/static/js/bundle.js:46:23
webpackJsonpCallback@http://localhost:3000/static/js/bundle.js:33:19
@http://localhost:3000/static/js/main.chunk.js:1:77
```

**Pull request merged**